pkgs/stdenv/generic/setup.sh

```diff
@@ -68,7 +68,8 @@ _callImplicitHook() {
     local hookName="$2"
     if declare -F "$hookName" > /dev/null; then
         "$hookName"
-    elif type -p "$hookName" > /dev/null; then
+    elif test -e "$hookName"; then
```

Comment:
Perhaps
```suggestion
    elif [ -e "$hookName" ]; then
```
to not mix `test` and `[` (below)

Reply:
If you don't mind, I've taken the liberty to use `[[` and I've also replaced the one below from `[` to `[[`

Reply:
I've switched the base to staging. I think targetting master for faster builds was just confusing to most people.